lib/resolver/index.js

```diff
@@ -0,0 +1,226 @@
+const path = require("path");
+const url = require("url");
+const os = require("os");
+const platform = os.platform();
+
+const SourceType = {
+  Local: 0,
+  URL: 1,
+  Package: 2,
+};
+
+exports.SourceType = SourceType;
+
+
+function resolve(fromSourcePath, toSourcePath, args, sourcePathType = SourceType.Local) {
+  return sourcePathType === SourceType.URL
+    ? resolveFromUrl(fromSourcePath, toSourcePath, args)
+    : resolveFromLocal(fromSourcePath, toSourcePath, args);
+}
+
+exports.resolve = resolve;
+
+class Entry {
+  constructor(file, type = SourceType.Local, packageFolder = "") {
+    this.type = type;
+    if (type === SourceType.Local) {
+      // file relative to packageStart
+      this.file = path.basename(file);
+      // package folder
+      this.baseDir = path.dirname(file);
+      this.url = null;
+    } else if (type === SourceType.URL) {
+      this.file = null;
+      this.baseDir = null;
+      this.url = url.parse(file);
+    } else { // package
+      this.file = file;
+      this.baseDir = packageFolder;
+      this.url = null;
+    }
+  }
+
+  resolveSync(readFileSync, fetchSync) {
+    if (this.type === SourceType.Package) {
+      // resolve it as a package
+      const moduleFolder = this.baseDir;
+      const fileRelativePath = this.file;
+      const packageJsonPath = path.join(moduleFolder, "package.json");
+      const pkg = readFileSync(packageJsonPath);
+      let ascFolder = "assembly";
```

Comment:
I guess better move "assembly" out to top level const and call something like `ASC_DEFAULT_FOLDER = "assembly"`